### PR TITLE
POM v1

### PR DESCRIPTION
**Pull request by JuanManu59**

### CODE REVIEW

File: cypress/page/login.page.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  private loginBtn: string;
+
+  constructor() {
+    this.addressText = "#email";
+    this.passwordText = "#passwd";
+    this.loginBtn = "button#SubmitLogin.button.btn.btn-default.button-medium";
+  }
+
+  public writeAddress(): void {
+    cy.get(this.addressText).type("aperdomobo@gmail.com");
```
Comment (AlejaGonzalezV) [inline cypress/page/login.page.ts]: don't use the mail and password values directly, add them as method parameters and add them in the test file when you call loginPage.login(mail, pwd);

File: cypress/page/payment-step.page.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    cy.get(this.payByBankBtn).click();
+  }
+
+  public proceedCheckout(): void {
+    cy.get(this.checkoutBtn).click();
+  }
+}
+
+export {PaymentStepPage};
+
```
Comment (AlejaGonzalezV) [inline cypress/page/payment-step.page.ts]: delete this line, just leave one empty line at the end of the file

File: cypress/page/shipping-step.page.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    cy.get(this.checkTermsOfService).click();
+  }
+
+  public proceedCheckout(): void {
+    cy.get(this.checkoutBtn).click();
+  }
+}
+
+export {ShippingStepPage};
+
```
Comment (AlejaGonzalezV) [inline cypress/page/shipping-step.page.ts]: delete this line, just leave one empty line at the end of the file

File: cypress/page/shoping-cart.page.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    this.shoppingMenu = ".cart_navigation span";
+  }
+
+  public goToCheckout(): void {
+    cy.get(this.shoppingMenu).click();
+  }
+}
+
+export {ShopingCartPage};
+
```
Comment (AlejaGonzalezV) [inline cypress/page/shoping-cart.page.ts]: delete this line, just leave one empty line at the end of the file

File: cypress/page/login.page.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    cy.get(this.passwordText).type("WorkshopProtractor");
+  }
+
+  public login(): void {
+    cy.get(this.loginBtn).click();
+  }
+}
+
+export {LoginPage};
+
```
Comment (AlejaGonzalezV) [inline cypress/page/login.page.ts]: delete this line, just leave one empty line at the end of the file

File: cypress/page/address-step.page.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    this.proceedCheckout = "button.button.btn.btn-default.button-medium";
+  }
+
+  public proceedCheckoutAuth(): void {
+    cy.get(this.proceedCheckout).click();
+  }
+}
+
+export {AddressStepPage};
+
```
Comment (AlejaGonzalezV) [inline cypress/page/address-step.page.ts]: delete this line, just leave one empty line at the end of the file

File: cypress/e2e/buy-shirt.cy.ts [hunk 7]
```diff
@@ -1,20 +1,42 @@
...
+    addressStepPage.proceedCheckoutAuth();
+
+    shippingStepPage.checkToS();
+    shippingStepPage.proceedCheckout();
+
+    paymentStepPage.payWithBank();
+    paymentStepPage.proceedCheckout();
 
     cy.get("#center_column > div > p > strong")
         .should("have.text", "Your order on My Store is complete.");
```
Comment (AlejaGonzalezV) [inline cypress/e2e/buy-shirt.cy.ts]: you can add this into the pom where it is visible

File: cypress/page/login.page.ts (hunk 1 above)
Comment (JuanManu59) [reply cypress/page/login.page.ts]: resolved

File: cypress/page/payment-step.page.ts (hunk 2 above)
Comment (JuanManu59) [reply cypress/page/payment-step.page.ts]: resolved

File: cypress/page/shipping-step.page.ts (hunk 3 above)
Comment (JuanManu59) [reply cypress/page/shipping-step.page.ts]: resolved

File: cypress/page/shoping-cart.page.ts (hunk 4 above)
Comment (JuanManu59) [reply cypress/page/shoping-cart.page.ts]: resolved

File: cypress/page/login.page.ts (hunk 5 above)
Comment (JuanManu59) [reply cypress/page/login.page.ts]: resolved

File: cypress/page/address-step.page.ts (hunk 6 above)
Comment (JuanManu59) [reply cypress/page/address-step.page.ts]: resolved

File: cypress/e2e/buy-shirt.cy.ts (hunk 7 above)
Comment (JuanManu59) [reply cypress/e2e/buy-shirt.cy.ts]: resolved

**Review by AlejaGonzalezV — DISMISSED** [review]

File: cypress/e2e/buy-shirt.cy.ts [hunk 8]
```diff
@@ -1,22 +1,44 @@
...
-    cy.get("a.bankwire").click();
-    cy.get("button.button.btn.btn-default.button-medium").click();
-
-    cy.get("#center_column > div > p > strong")
-        .should("have.text", "Your order on My Store is complete.");
+    menuContentPage.visitMenuContentPage();
+
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.addToMyCar();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: typo: addToMyCar***t***

File: cypress/e2e/buy-shirt.cy.ts [hunk 9]
```diff
@@ -1,22 +1,44 @@
...
+
+    addressStepPage.proceedCheckoutAuth();
+
+    shippingStepPage.checkToS();
+    shippingStepPage.proceedCheckout();
+
+    paymentStepPage.payWithBank();
+    paymentStepPage.proceedCheckout();
+
+    paymentStepPage.verifyOrderComplete();
```
Comment (holgiosalos) [inline cypress/e2e/buy-shirt.cy.ts]: in this case I would receive the text you want to verify as parameter, in that way the page object is not attached to an specific message

File: cypress/e2e/buy-shirt.cy.ts (hunk 8 above)
Comment (JuanManu59) [reply cypress/e2e/buy-shirt.cy.ts]: resolved

File: cypress/e2e/buy-shirt.cy.ts (hunk 9 above)
Comment (JuanManu59) [reply cypress/e2e/buy-shirt.cy.ts]: resolved

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**